### PR TITLE
some small changes

### PR DESCRIPTION
**Pull request by esohns**
- added a byte-swap macro for ACE_UINT64
- allow ACE_Module to be sub-classed
- allow SOCK_Dgram_Mcast to be sub-classed

### CODE REVIEW

File: ACE/ace/Basic_Types.h [hunk 1]
```diff
@@ -285,9 +285,11 @@ ACE_END_VERSIONED_NAMESPACE_DECL
 // Byte swapping macros to deal with differences between little endian
 // and big endian machines.  Note that "long" here refers to 32 bit
 // quantities.
+# define ACE_SWAP_LONG_LONG(LL) ((ACE_SWAP_LONG ((LL)&0xFFFFFFFF) << 32) \
```
Comment (jwillemsen) [inline ACE/ace/Basic_Types.h:288]: We should have a test extension for this new macro

File: ACE/ace/Module.h [hunk 2]
```diff
@@ -59,6 +59,8 @@ class ACE_Export ACE_Module_Base
     /// Indicates that close() should not delete any Tasks.
     M_DELETE_NONE = 4
   };
+
+  inline virtual ~ACE_Module_Base (void) {}
```
Comment (jwillemsen) [inline ACE/ace/Module.h:63]: Make use of C++11 default, remove redundant void

File: ACE/ace/Module.h [hunk 3]
```diff
@@ -158,7 +160,7 @@ class ACE_Module : public ACE_Module_Base
   void link (ACE_Module<ACE_SYNCH_USE, TIME_POLICY> *m);
 
   /// Get the next pointer to the module above in the stream.
-  ACE_Module<ACE_SYNCH_USE, TIME_POLICY> *next ();
+  virtual ACE_Module<ACE_SYNCH_USE, TIME_POLICY> *next (void);
```
Comment (jwillemsen) [inline ACE/ace/Module.h:163]: Remove redundant void

File: ACE/ace/SOCK_Dgram_Mcast.h [hunk 4]
```diff
@@ -1,407 +1,407 @@
...
-
-ACE_END_VERSIONED_NAMESPACE_DECL
-
-#if defined (__ACE_INLINE__)
-#include "ace/SOCK_Dgram_Mcast.inl"
-#endif /* __ACE_INLINE__ */
-
-#include /**/ "ace/post.h"
-#endif /* ACE_SOCK_DGRAM_MCAST_H */
+// -*- C++ -*-
```
Comment (jwillemsen) [inline ACE/ace/SOCK_Dgram_Mcast.h:1]: Github shows this file as 100% changed, please check that, now it can't be reviewed

File: ACE/ace/Task_T.h [hunk 5]
```diff
@@ -124,7 +124,7 @@ class ACE_Task : public ACE_Task_Base
 
   // = Pointers to next ACE_Task_Base (if ACE is part of an ACE_Stream).
   /// Get next Task pointer.
-  ACE_Task<ACE_SYNCH_USE, TIME_POLICY> *next ();
+  virtual ACE_Task<ACE_SYNCH_USE, TIME_POLICY> *next (void);
```
Comment (jwillemsen) [inline ACE/ace/Task_T.h:127]: Remove redundant void

**Review by jwillemsen — CHANGES_REQUESTED** [review]

File: ACE/ace/Basic_Types.h (hunk 1 above)
Comment (esohns) [reply ACE/ace/Basic_Types.h:288]: OK, which test should I extend ? I searched in /tests but could not find any usage of the other "ACE_SWAP_x"-macros there

File: ACE/ace/Module.h (hunk 2 above)
Comment (esohns) [reply ACE/ace/Module.h]: Done

File: ACE/ace/Module.h (hunk 3 above)
Comment (esohns) [reply ACE/ace/Module.h]: Done

File: ACE/ace/SOCK_Dgram_Mcast.h (hunk 4 above)
Comment (esohns) [reply ACE/ace/SOCK_Dgram_Mcast.h:1]: I resubmitted the file

File: ACE/ace/Task_T.h (hunk 5 above)
Comment (esohns) [reply ACE/ace/Task_T.h]: Done

File: ACE/ace/Basic_Types.h (hunk 1 above)
Comment (jwillemsen) [reply ACE/ace/Basic_Types.h:288]: Maybe Basic_Types_Test.cpp

File: ACE/ace/Module.h [hunk 6]
```diff
@@ -59,6 +59,8 @@ class ACE_Export ACE_Module_Base
     /// Indicates that close() should not delete any Tasks.
     M_DELETE_NONE = 4
   };
+
+  inline virtual ~ACE_Module_Base () {}
```
Comment (jwillemsen) [inline ACE/ace/Module.h:63]: Use `default`

File: ACE/ace/SOCK_Dgram_Mcast.h [hunk 7]
```diff
@@ -376,15 +376,15 @@ class ACE_Export ACE_SOCK_Dgram_Mcast : public ACE_SOCK_Dgram
   /// Empty the dynamic subscription list.
   int clear_subs_list ();
 
+  /// Multicast address to which local send() methods send datagrams.
```
Comment (jwillemsen) [inline ACE/ace/SOCK_Dgram_Mcast.h:379]: Don't reorder member declarations as that will trigger compile warnings on the ordering of these members in the constructors

File: ACE/ace/Basic_Types.h (hunk 1 above)
Comment (esohns) [reply ACE/ace/Basic_Types.h:288]: Done

File: ACE/ace/Module.h (hunk 6 above)
Comment (esohns) [reply ACE/ace/Module.h]: Done

File: ACE/ace/SOCK_Dgram_Mcast.h (hunk 7 above)
Comment (esohns) [reply ACE/ace/SOCK_Dgram_Mcast.h:379]: I had to, because I moved two members into the 'protected' area. Updated ctor in SOCK_Dgram_Mcast.cpp

**Review by esohns — COMMENTED** [review]
please review the proposed changes

File: ACE/tests/Basic_Types_Test.cpp [hunk 8]
```diff
@@ -33,8 +33,8 @@
 using a_function_pointer = void *(*)(void *);
 
 static
-u_int
-check (const ACE_TCHAR *message, u_int i, u_int j)
+ACE_UINT32
+check (const ACE_TCHAR* message, ACE_UINT32 i, ACE_UINT32 j)
```
Comment (jwillemsen) [inline ACE/tests/Basic_Types_Test.cpp:37]: check32 as function name? Can't the static be removed, also check64

**Review by jwillemsen — APPROVED** [review]

**Comment by jwillemsen** [comment]
Why closed this, spend time today on this, want to make one last change on Monday and wanted to merge it when the CI is ok

**Comment by jwillemsen** [comment]
Thanks!

**Pull request merged**